backend/app/api/endpoints/auth.py

```
@@ -53,3 +64,28 @@ def login(db: Session = Depends(get_db), login_data: LoginRequest = Body(...)):
+) -> AdminPasswordSetupStatusResponse:
+    """Return whether first-run admin password setup is required."""
+    return AdminPasswordSetupStatusResponse(
+        required=is_admin_password_setup_required(db),
+        admin_username=INITIAL_ADMIN_USERNAME,
+    )
+
+
+@router.post("/admin-password/setup", response_model=Token)
+def setup_admin_password(
+    setup_data: AdminPasswordSetupRequest,
+    db: Session = Depends(get_db),
```

Ruff (0.15.17)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)